### PR TITLE
OnRender type of functional prop for column-divider in DetailsList

### PR DESCRIPTION
**Pull request by s-vaibhav**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

A functional prop (onRenderDivider) added to DetailsList component which can be used to wrap/override the default behaviour of column header divider.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts [hunk 1]
```diff
@@ -383,6 +383,11 @@ export interface IColumn {
    */
   onRender?: (item?: any, index?: number, column?: IColumn) => any;
 
+  /**
+   * can be used to wrap the existing functionality of column divider
```
Comment (shagra-ms) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts]: >can be used to wrap the existing functionality of column divider [](start = 5, length = 64)

Use language similar to other onRender* props

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts [hunk 2]
```diff
@@ -383,6 +383,11 @@ export interface IColumn {
    */
   onRender?: (item?: any, index?: number, column?: IColumn) => any;
 
+  /**
+   * If provider, can be used to render a custom column header divider
+   */
+  onRenderDivider?: (columnIndex: number, defaultRenderer: Function) => any;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts]: Our onRender functions typically use the IRenderFunction which provides the user with the original render function to be used in their on onRender. Is there a reason this wasn't used?

**Comment by micahgodbolt** [comment]
@ThomasMichon can you take a quick look at this change?

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts (hunk 2 above)
Comment (s-vaibhav) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts]: Updated the definition.

**Pull request merged**